homeassistant/components/nice_go/cover.py

```diff
@@ -84,6 +85,24 @@ async def async_close_cover(self, **kwargs: Any) -> None:
                 translation_key="close_cover_error",
                 translation_placeholders={"exception": str(err)},
             ) from err
+        except AuthFailedError:
+            # Try refreshing token and retry
+            try:
+                await self.coordinator.update_refresh_token()
+                await self.coordinator.api.close_barrier(self._device_id)
+            except (ApiError, ClientError) as err:
+                raise HomeAssistantError(
```

Comment:
Rather make this a decorator so you can easily execute the same function twice without code duplication

Reply:
Don't know how that didn't cross my mind!